onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -93,67 +128,39 @@ inline void CopyPrim(uint8_t* target, const uint8_t* source) {
   *reinterpret_cast<T*>(target) = *reinterpret_cast<const T*>(source);
 }
 
+// The function does not check num_axes > 0 but this is expected.
+template <class T>
+static void TypedDoTransposeEltWise(int64_t num_axes, const std::vector<int64_t>& target_dims, size_t num_blocks,
+                                    const std::vector<size_t>& stride, const uint8_t* source, uint8_t* target) {
+  std::vector<MultiIndex> mindex(num_axes);
+  size_t naxes = IncrementIndexAndComputeOffsetSetup(mindex.data(), num_axes, target_dims, stride, sizeof(T));
+
+  const uint8_t* local_source = source;
+  for (size_t i = 0; i < num_blocks; ++i) {
+    CopyPrim<T>(target, local_source);
+    IncrementIndexAndComputeOffset(mindex.data(), naxes, local_source);
+    target += sizeof(T);
```

Comment:
If I understand correctly, target should be T*.  
And the for loop would be like:
```c++
T* target_end = target+num_blocks;
for(T* p=target; p!=target_end;++p){
   ....
}
```

Reply:
Yes, we could save another variable.